scripts/createNewComponent.js

```diff
@@ -0,0 +1,76 @@
+import fs from 'fs';
+import chalk from 'chalk';
+import nextTask from './nextTask.js';
+
+const componentName = process.argv[2];
+
+if (!componentName) {
+  console.error(`${chalk.red('✘')} Please provide a component name.`);
+  process.exit(1);
+}
+
+const folderName = componentName.replace(/[A-Z]/g, (match) => `-${match.toLowerCase()}`);
+
+const createFolder = () => {
+  const folderPath = `src/components/${folderName}`;
+  if (!fs.existsSync(folderPath)) {
+    fs.mkdirSync(folderPath, { recursive: true });
+  } else {
+    console.error(`${chalk.red('✘')} Folder already exists.`);
+    process.exit(1);
+  }
+  fs.mkdirSync(`src/components/${folderName}`, { recursive: true });
+};
+
+const createComponentFile = () => {
+  const className = componentName
+    .split('-')
+    .map((n) => n.charAt(0).toUpperCase() + n.slice(1))
+    .join('');
+  const content = `
+    import { html, LitElement } from 'lit';
+    import { customElement } from 'lit/decorators.js';
+    import styles from './${componentName}.styles';
+
+    @customElement('${componentName}')
+    export default class ${className} extends LitElement {
+      constructor() {
+        super();
+      }
+    }
+    
+    declare global {
+      interface HTMLElementTagNameMap {
+        '${componentName}': ${className};
+      }
+    }`;
+
+  fs.writeFileSync(`src/components/${folderName}/${componentName}.component.ts`, content);
+};
+
+const createDemoFile = () => {
+  const demoContent = `
+  import { html } from 'lit';
+
+  export default html\`\``;
+
+  fs.writeFileSync(`src/components/${folderName}/${componentName}.demo.ts`, demoContent);
+};
+
+const createStylesFile = () => {
+  const stylesContent = `
+  import { css } from 'lit';
+
+  export default css\`\``;
+
+  fs.writeFileSync(`src/components/${folderName}/${componentName}.styles.ts`, stylesContent);
+};
+
+const createFiles = () => {
+  createComponentFile();
+  createDemoFile();
+  createStylesFile();
```

Comment:
Hadde du fått til å legge inn export-linja i src/components/index.ts også?
Hvis ikke hadde det vært fint med en påminnelse i konsollet om å gjøre det.

Samme med main.ts. Kanskje vanskelig å legge inn kode automatisk her, men en påminnelse i konsollet hadde vært fint!

Reply:
er det ikke bedre å legge til en kommentar i en compoonen.ts fila (som man må slette eller blir den irriterende :P) at du må gjerne starte med å legge importer i både main.ts og index? Jeg har ikke gjørt det automatisk fordi kan være at man vil slette mappe (hvis man gjorde noe feil), og da import blir igjen i noen av de to filene. Siden vi også pleier å legge til imports alfabetisk jeg vil ikke skrive om hele filen via script.